--- a/src/supertux/main.cpp
+++ b/src/supertux/main.cpp
@@ -23,6 +23,9 @@
 
 #include <SDL_image.h>
 #include <SDL_ttf.h>
+#ifdef __ANDROID__
+#include <SDL_system.h>
+#endif
 #include <physfs.h>
 #include <tinygettext/log.hpp>
 #include <fmt/format.h>
@@ -180,7 +183,16 @@ PhysfsSubsystem::PhysfsSubsystem(const char* argv0,
   m_datadir(),
   m_userdir()
 {
-  if (!PHYSFS_init(argv0))
+#ifdef __ANDROID__
+  PHYSFS_AndroidInit physfsAndroidInit{};
+  physfsAndroidInit.jnienv = SDL_AndroidGetJNIEnv();
+  physfsAndroidInit.context = SDL_AndroidGetActivity();
+  const char* physfs_init_arg = reinterpret_cast<const char*>(&physfsAndroidInit);
+#else
+  const char* physfs_init_arg = argv0;
+#endif
+  if (!PHYSFS_init(physfs_init_arg))
   {
     std::stringstream msg;
     msg << "Couldn't initialize physfs: " << physfsutil::get_last_error();
